File: app/src/main/java/com/example/firefighters/viewmodels/MessageViewModel.java
package com.example.firefighters.viewmodels;

import android.app.Activity;

import com.example.firefighters.models.MessageModel;
import com.example.firefighters.models.MessageModel;
import com.example.firefighters.models.UnitModel;
import com.example.firefighters.repositories.MessageRepository;
import com.google.firebase.firestore.DocumentSnapshot;
import com.google.firebase.firestore.QuerySnapshot;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.ViewModel;

public class MessageViewModel extends ViewModel {

    private MessageRepository repository;

    public void init() {
        if (repository != null)
            return;
        repository = MessageRepository.getInstance();
    }

    public LiveData<MessageModel> loadMessageModel(int id) {
        return repository.loadMessageModel(id);
    }
    public LiveData<String> saveDataToStorage(String parent, String path) {
        return repository.saveDataToStorage(parent, path);
    }
    public LiveData<Integer> saveMessage(MessageModel messageModel ) {
        return repository.saveMessage(messageModel);
    }
    public LiveData<Integer> deleteMessage(int id) {
        return repository.deleteMessage(id);
    }
    public LiveData<Integer> updateMessage(MessageModel messageModel ) {
        return repository.updateMessage(messageModel);
    }
}

File: app/src/main/java/com/example/firefighters/viewmodels/models/MessageViewModel.kt
package com.example.firefighters.viewmodels.models

import androidx.lifecycle.LiveData
import androidx.lifecycle.ViewModel
import androidx.lifecycle.ViewModelProvider
import com.example.firefighters.models.MessageModel
import com.example.firefighters.repositories.MessageRepository

class MessageViewModel(private val repository: MessageRepository) : ViewModel() {

    fun loadMessageModel(id: Int): LiveData<MessageModel?> {
        return repository.loadMessageModel(id)
    }

    fun saveDataToStorage(parent: String?, path: String?): LiveData<String?> {
        return repository.saveDataToStorage(parent!!, path)
    }

    fun saveMessage(messageModel: MessageModel?): LiveData<Int> {
        return repository.saveMessage(messageModel!!)
    }

    fun deleteMessage(id: Int): LiveData<Int> {
        return repository.deleteMessage(id)
    }

    fun updateMessage(messageModel: MessageModel?): LiveData<Int> {
        return repository.updateMessage(messageModel!!)
    }

    class Factory(
        private val repository: MessageRepository
    ) : ViewModelProvider.NewInstanceFactory() {
        override fun <T : ViewModel> create(modelClass: Class<T>): T {
            return MessageViewModel(repository) as T
        }
    }
}
